### ISSUE TITLE
Arc Rendering Bugs for Nearly Circlular Arc with Large Corner Radius

### ISSUE THREAD

**Issue by michaelkaplanworkday** [mbostock/d3#2249]
![arcbugs](https://cloud.githubusercontent.com/assets/962036/6499166/58d405d8-c2ab-11e4-9aee-495898509862.png)

[Online example of bug](http://bl.ocks.org/michaelkaplanworkday/d181cb1ce1f65983de73)
[Previous related issue](https://github.com/mbostock/d3/issues/1131) from bm-w
[Previous related issue has attached example](http://bl.ocks.org/bm-w/5124918)

Through debugging of d3.js file, I found that svg/arc.js Lines 123-127 and Lines 145-149 are causing this issue and the issue does not occur if the else blocks are removed and the "if" cases (Lines 117-112 and 139-144) are always used (remove the "if" check). I'm unsure if the else cases need to be there. If not, then they can be removed but if they do need to be there then perhaps the solution proposed by deanmalmgren involving automatically reducing the radius in these corner cases would work best.

Excerpt of lines causing the bug: from [svg/arc.js](https://github.com/mbostock/d3/blob/master/src/svg/arc.js)
```javascript
path.push("M", t30[0], "A", rc1, ",", rc1, " 0 1,", cr, " ", t12[0]);
```

```javascript
path.push("L", t21[0], "A", rc0, ",", rc0, " 0 0,", cr, " ", t03[0]);
```

**Comment by rafeequali** [comment]
Hello..
Any fixes for this issue ?

**Comment by mbostock** [comment]
Fixed in 2414655c81e6408420a232dd59f8d121abc4757f, merged into #2591 for 3.5.7.

Related #2592.

**Comment by mbostock** [comment]
![screen shot 2015-10-23 at 2 08 55 pm](https://cloud.githubusercontent.com/assets/230541/10705015/a58496c2-798f-11e5-9057-28ddf0f5a580.png)
